pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularMedicalReleaseV2.java

```diff
@@ -0,0 +1,283 @@
+                .map(block -> (QuestionBlockDef) block)
+                .filter(block -> block.getQuestion().getStableId().equals(questionStableIdToRemove))
+                .findFirst()
+                .orElseThrow(() -> new DDPException("Could not find block for stable id:" + questionStableIdToRemove));
+
+        QuestionDef questionToReplaceAndDelete = v2QuestionBlockToUpdate.getQuestion();
+
+        FormActivityDef activityDefV1 = findActivityDef(studyId, activityCode, oldVersionTag, handle);
+
+        QuestionDef questionToAttachToActivityV2 = activityDefV1.getQuestionByStableId(questionStableIdToUse);
+
+        assignQuestionToBlock(v2QuestionBlockToUpdate, questionToAttachToActivityV2, handle);
+
+        // The original question from v2 is not needed anymore. Let's do some cleanup
+        QuestionDao questionDao = handle.attach(QuestionDao.class);
+        questionDao.deleteQuestion(questionToReplaceAndDelete);
+
+        // We are checking our work now
+        Long originalQuestionId = findActivityDef(studyId, activityCode, oldVersionTag, handle)
+                .getQuestionByStableId(questionStableIdToUse).getQuestionId();
+        FormActivityDef activityDefV2AfterQuestionUpdate = findActivityDef(studyId, activityCode, "v2", handle);
+        Long questionIdInV2 = activityDefV2AfterQuestionUpdate.getQuestionByStableId(questionStableIdToUse).getQuestionId();
+        if (!originalQuestionId.equals(questionIdInV2)) {
+            throw new DDPException("Update of question did not work");
+        }
+        QuestionDef questionThatShouldHaveBeenDeleted = activityDefV2AfterQuestionUpdate.getQuestionByStableId(questionStableIdToRemove);
+        if (questionThatShouldHaveBeenDeleted != null) {
+            throw new DDPException("New activity still hanging on to question that should have been removed");
+        }
+    }
+
+    private int assignQuestionToBlock(QuestionBlockDef questionBlockToUpdate, QuestionDef questionToPreserve, Handle handle) {
+        return handle.execute("update block__question set question_id=? where block_id=?",
+                questionToPreserve.getQuestionId(), questionBlockToUpdate.getBlockId());
+    }
+
+    private FormActivityDef findActivityDef(long studyId, String activityCode, String versionTag, Handle handle) {
+        ActivityVersionDto originalV1VersionDto = handle.attach(JdbiActivityVersion.class)
+                .findByActivityCodeAndVersionTag(studyId, activityCode, versionTag)
+                .orElseThrow(() -> new DDPException("Could not find activity version"));
+        JdbiActivity jdbiActivity = handle.attach(JdbiActivity.class);
+        Optional<ActivityDto> activityDto = jdbiActivity.findActivityByStudyIdAndCode(studyId, activityCode);
+        if (activityDto.isEmpty()) {
+            throw new DDPException("Could not find activity with study id: " + studyId + " and code:" + activityCode);
+        }
```

Comment:
This could be rewritten as the following to get rid of the `isEmpty()` check
```suggestion
        var activityDto = jdbiActivity.findActivityByStudyIdAndCode(studyId, activityCode);
                .orElseThrow(() -> {
                    return new DDPException("Could not find activity with study id: " + studyId + " and code:" + activityCode)
                });
```

Reply:
That is nicer. THanks! 3888327b1